CorsixTH/Lua/dialogs/patient.lua

```diff
@@ -114,6 +114,20 @@ function UIPatient.normaliseWarmth(warmth)
   return warmth
 end
 
+--! Draw a bar in the patient dialogue window.
+--!param canvas Canvas to draw at.
+--!param xbase Horizontal base position.
+--!param ybase Vertical base position.
+--!param xpos Horizontal offset.
+--!param ypos Vertical offset.
+--!param value Fraction to draw.
+function UIPatient:drawBar(canvas, xbase, ybase, xpos, ypos, value)
+  local width = math.floor(value * 40 + 0.5)
+  for dx = 0, width - 1 do
```

Comment:
I think you still need to handle the 0 case here, otherwise math.floor(0 * 40 + 0.5) == 0, then you subtract 1, so you would be saying from 0 to -1 step 1, which seems like a bad thing to do.